### PR TITLE
[Compute] vmss create/update: Add --scale-in-policy

### PR DESCRIPTION
**Pull request by qwordy**
Issue #10949

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 1]
```diff
@@ -1875,6 +1875,23 @@ def test_vmss_create_and_modify(self):
...
 
+    @ResourceGroupPreparer(name_prefix='cli_test_vmss_scale_in_policy_')
+    def test_vmss_scale_in_policy(self, resource_group):
+        self.kwargs.update({
+            'vmss': 'vmss1'
+        })
+        self.cmd('vmss create -g {rg} -n {vmss} --image centos --scale-in-policy NewestVM', checks=[
+            self.check('vmss.scaleInPolicy.rules[0]', 'NewestVM')
+        ])
+        self.cmd('vmss update -g {rg} -n {vmss} --set scaleInPolicy.rules[0]=OldestVM', checks=[
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: `rules` is a field internal to [`ScaleInPolicy`](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachinescalesets/createorupdate#scaleinpolicy). I prefer implementing a strongly-typed argument of `--scale-in-policy` in `vmss update`.

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 2]
```diff
@@ -1875,6 +1875,23 @@ def test_vmss_create_and_modify(self):
...
+            self.check('scaleInPolicy.rules[0]', 'OldestVM')
+        ])
+        self.cmd('vmss update -g {rg} -n {vmss} --set automaticRepairsPolicy.enabled=false automaticRepairsPolicy.gracePeriod=PT5M automaticRepairsPolicy.maxInstanceRepairsPercent=100', checks=[
+            self.check('automaticRepairsPolicy.enabled', False),
+            self.check('automaticRepairsPolicy.gracePeriod', 'PT5M'),
+            self.check('automaticRepairsPolicy.maxInstanceRepairsPercent', 100)
+        ])
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: This test is unrelated to this PR. I prefer removing it, moving to another PR or clarifying it in PR's description.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 3]
```diff
@@ -506,6 +506,7 @@ def load_arguments(self, _):
         c.argument('computer_name_prefix', help='Computer name prefix for all of the virtual machines in the scale set. Computer name prefixes must be 1 to 15 characters long')
         c.argument('orchestration_mode', help='Choose how virtual machines are managed by the scale set. In VM mode, you manually create and add a virtual machine of any configuration to the scale set. In ScaleSetVM mode, you define a virtual machine model and Azure will generate identical instances based on that model.',
                    arg_type=get_enum_type(['VM', 'ScaleSetVM']), is_preview=True)
+        c.argument('scale_in_policy', nargs='+', arg_type=get_enum_type(self.get_models('VirtualMachineScaleSetScaleInRules')), help='Specify the scale-in policy (space delimited) that decides which virtual machines are chosen for removal when a Virtual Machine Scale Set is scaled-in.')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: Can we update scale_in_policy property in update command?

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: Removed

**Comment by qwordy** [comment]
OK. I will add --scale-in-policy to update command.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 4]
```diff
@@ -86,6 +86,11 @@ def load_arguments(self, _):
         arg_type=get_three_state_flag(), min_api='2018-06-01',
         help='Enables or disables the capability to have 1 or more managed data disks with UltraSSD_LRS storage account')
 
+    scale_in_policy_type = CLIArgumentType(
+        nargs='+', arg_type=get_enum_type(self.get_models('VirtualMachineScaleSetScaleInRules')),
+        help='Specify the scale-in policy (space delimited) that decides which virtual machines are chosen for removal when a Virtual Machine Scale Set is scaled-in.'
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: I prefer using "specifies" rather than "decides".

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: I copy it from swagger
